docs/guides/03_writing_queries.md

```diff
@@ -0,0 +1,68 @@
+# Writing queries
+
+## Intro
+
+In the previous guide, we described how to write commands - actions which modify the data in the system. This guide shows how queries can be written. Contrary to commands, queries are responsible for retrieving data without modifying any objects.
+
+Each query consists of two parts:
+
+* Contract
+* Query handler
+
+This is analogous to how commands are implemented, except a validator.
+
+## Contract
+
+As it is the case with command contracts, a query contract defines the parameters passed to a query as well as the structure of the response. We will show a simple query for retrieving a list of projects.
+
+```csharp
+[AllowUnauthorized]
+public class AllProjects : IQuery<List<ProjectDTO>>
+{
+    public bool SortByDescending { get; set; }
+}
+
+public class ProjectDTO
+{
+    public string Id { get; set; }
+    public string Name { get; set; }
+}
+```
+
+A contract for a query is defined by creating a class implementing the `IQuery<T>` interface. The `T` is the result type of a query which is serialized in the HTTP response after a query has been executed. Also, web and mobile clients can use the Contracts Generator to generate classes corresponding to defined queries and their result types.
+
+Fields in a query class represent parameters which can be passed in a request to alter the query results. In the above example there is a single query parameter of boolean type `SortByDescending` which will be used to determine sorting order of the results.
+
+## Handler
+
+A handler for a command is a class implementing `IQueryHandler<TContext, TQuery, TResult>` interface where `TQuery` is the contract type and `TResult` is the result type as defined in the contract. In our example, a query handler might look like this:
```

Comment:
We silently ignore what `TContext` is in here (and in command docs). I believe they should be explained at least briefly